### PR TITLE
[ML-DataFrame] rename jobs to transforms

### PR DESCRIPTION
**Pull request by hendrikmuhs**
rename jobs to transforms

Note: This PR only contains renamings, but 1 slight addition to make tests pass: remove type from the bulk index requests for rest test cases

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
@elasticmachine run elasticsearch-ci-1

**Comment by hendrikmuhs** [comment]
run gradle build tests 1

**Comment by hendrikmuhs** [comment]
run gradle build tests 2

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/GetDataFrameTransformsStatsAction.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public Request(StreamInput in) throws IOException {
 
         @Override
         public boolean match(Task task) {
-            // If we are retrieving all the jobs, the task description does not contain the id
+            // If we are retrieving all the s, the task description does not contain the id
```
Comment (davidkyle) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/GetDataFrameTransformsStatsAction.java]: `If we are retrieving all the s`

**Review by davidkyle — APPROVED** [review]
LGTM. I found one typo please don't ask for a second review

**Comment by hendrikmuhs** [comment]
run gradle build tests 2

**Comment by hendrikmuhs** [comment]
run gradle build tests 2

**Comment by hendrikmuhs** [comment]
run gradle build tests 2

**Comment by hendrikmuhs** [comment]
run gradle build tests 1

**Comment by hendrikmuhs** [comment]
run gradle build tests 1

**Pull request merged**